File: backend/src/main/scala/microservices/operations/attraction/objects/CreateAttractionPlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateAttractionPlannerRequest(managerId: String, attractionName: String, city: String, location: String, description: String, imageUrl: Option[String])
object CreateAttractionPlannerRequest:
  given sourceEncoder: Encoder[CreateAttractionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateAttractionPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/attraction/objects/CreateAttractionPlannerRequest.ts
// 本文件定义 CreateAttractionPlanner 的景点创建请求，供管理端创建景点时使用。

export type CreateAttractionPlannerRequest = {
  managerId: string
  attractionName: string
  city: string
  location: string
  description: string
  imageUrl?: string | null
}

export const createAttractionPlannerRequestFromJson = (json: string): CreateAttractionPlannerRequest =>
  JSON.parse(json) as CreateAttractionPlannerRequest

export const createAttractionPlannerRequestToJson = (value: CreateAttractionPlannerRequest): string =>
  JSON.stringify(value)
